Add Whitehall Attachment Content-Type headers

chrislo opened the pull request:
See: https://github.com/alphagov/asset-manager/issues/419

We are planning to change nginx so that requests for Whitehall attachments are served from Asset Manager instead of Whitehall. We've looked at all the existing attachments under `attachment_data` and requested examples of them to see what Content-Type header they are served with (#419). This PR registers the missing types so that the assets are served with the correct header.

floehopper commented on config/initializers/mime_types.rb:
```diff
@@ -4,9 +4,16 @@
+Mime::Type.register "application/vnd.openxmlformats-officedocument.present", :pptx
 Mime::Type.register "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", :xlsx
 Mime::Type.register "application/vnd.openxmlformats-officedocument.wordprocessingml.document", :docx
+Mime::Type.register "text/plain; charset=utf-8", :txt
```
Will this mean there will be a change in behaviour for Mainstream assets with a `txt` file extension?

Is this intentional? If so, I think it would be useful to highlight this in the commit note and explain what the content type used to be for this file extension and why we're happy to change it.

floehopper commented on config/initializers/mime_types.rb:
```diff
@@ -4,9 +4,16 @@
+Mime::Type.register "application/vnd.openxmlformats-officedocument.present", :pptx
 Mime::Type.register "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", :xlsx
 Mime::Type.register "application/vnd.openxmlformats-officedocument.wordprocessingml.document", :docx
+Mime::Type.register "text/plain; charset=utf-8", :txt
```
Will this mean there will be a change in behaviour for Mainstream assets with a `txt` file extension?

Is this intentional? If so, I think it would be useful to highlight this in the commit note and explain what the content type used to be for this file extension and why we're happy to change it.

floehopper replied:
Are there any other file extensions which will now have a different content type for Mainstream assets? If so, it would be good to make this clear in the commit note and explain why we're happy with the changes.

chrislo replied:
The change to `.txt` affects 3 non-whitehall assets. I've explained more in the commit note. 

The other file extensions affected are ones where Asset Manager would previously have returned `application/octet-stream`, which were incorrect. More detail of this is in the [linked comment](https://github.com/alphagov/asset-manager/issues/419#issuecomment-368575610). I don't think I'll count all of these up and list out which ones are affected now, as this change is an improvement (it sets the correct content-type).

chrislo commented:
Thanks @floehopper! I've addressed your feedback in commit notes, and added a new commit to add the 2 additional, unused mime-type registrations in whitehall.
